src/core/metadata/qgslayermetadataformatter.h

```diff
@@ -0,0 +1,80 @@
+/***************************************************************************
+                             qgslayermetadataformatter.h
+                             ---------------------------
+    begin                : September 2017
+    copyright            : (C) 2017 by Etienne Trimaille
+    email                : etienne dot trimaille at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSLAYERMETADATAFORMATTER_H
+#define QGSLAYERMETADATAFORMATTER_H
+
+#include "qgis.h"
+#include "qgis_core.h"
+#include "qgslayermetadata.h"
+
+/**
+ * \ingroup core
+ * \class QgsLayerMetadataFormatter
+ * \brief Class for metadata formatter.
+ * \since QGIS 3.0
+ */
+class CORE_EXPORT QgsLayerMetadataFormatter
+{
+  public:
+
+    /**
+     * Constructor for QgsLayerMetadataFormatter
+     */
+    QgsLayerMetadataFormatter( const QgsLayerMetadata &metadata );
+
+    /**
+     * Formats the "Access" section according to a \a metadata object.
+     * This will return a HTML table.
+     */
+    const QString accessSectionHtml();
```

Comment:
The `const` should be put at the end (`QString accessSectionHtml() const;`) we indicate that the method will not modifiy this formatter object. Const used like in the code here helps nothing, in theory it indicates that the returned string cannot be modified, but since it's a copy that's returned, the caller can just happily modify its copy anyway. In short, `const` used the way it's used here doesn't have any effect.

Reply:
Ahh, thanks. I was lost why `const` is sometimes on left or on the right. Gonna fix it